sources/main.cpp

```diff
@@ -180,62 +314,62 @@ main(int argc, char* argv[]) {
     }
     return EXIT_FAILURE;
   }
+
+  if (!boost::filesystem::exists(boost::filesystem::path(runtimeConfig.networkFilePath))) {
+    throw Error(NetworkFileNotFound, runtimeConfig.networkFilePath);
+  }
+  if (!runtimeConfig.contingenciesFilePath.empty() && !boost::filesystem::exists(boost::filesystem::path(runtimeConfig.contingenciesFilePath))) {
+    throw Error(ContingenciesFileNotFound, runtimeConfig.contingenciesFilePath);
+  }
+  if (userRequest == dfl::common::Options::Request::RUN_SIMULATION_NSA && runtimeConfig.contingenciesFilePath.empty()) {
+    throw Error(ContingenciesFileNotFound, runtimeConfig.contingenciesFilePath);
+  }
+
+  switch (userRequest) {
+  case dfl::common::Options::Request::RUN_SIMULATION_N:
+    LOG(info, SteadyStateInfo, runtimeConfig.networkFilePath, runtimeConfig.configPath);
+    break;
+  case dfl::common::Options::Request::RUN_SIMULATION_SA:
+    LOG(info, SecurityAnalysisInfo, runtimeConfig.networkFilePath, runtimeConfig.contingenciesFilePath, runtimeConfig.configPath);
+    break;
+  case dfl::common::Options::Request::RUN_SIMULATION_NSA:
+  default:
+    LOG(info, SteadyStateAndSecurityAnalysisInfo, runtimeConfig.networkFilePath, runtimeConfig.contingenciesFilePath, runtimeConfig.configPath);
+    break;
+  }
+
+  boost::filesystem::path parFilesDir(root);
+  parFilesDir.append("etc");
+
+  SimulationParams params;
+  params.setRuntimeConfig(&runtimeConfig)
+      .setMpiContext(&mpiContext)
+      .setTimeStart(timeStart)
+      .setParametersDirPath(parFilesDir)
+      .setResourcesDirPath(resourcesDir)
+      .setNetworkFilePath(runtimeConfig.networkFilePath)
+      .setLocale(locale);
+
+  if (userRequest == dfl::common::Options::Request::RUN_SIMULATION_N || userRequest == dfl::common::Options::Request::RUN_SIMULATION_NSA) {
+    params.setSimulationKind(dfl::inputs::Configuration::SimulationKind::STEADY_STATE_CALCULATION);
+
+    boost::shared_ptr<dfl::Context> context = buildContext(params);
+    execSimulation(context, params);
+  }
+
   DYNAlgorithms::mpi::Context::sync();
 
-  try {
-    auto timeSimuStart = std::chrono::steady_clock::now();
-    context->execute();
-    context->exportResults(true);
+  if (userRequest == dfl::common::Options::Request::RUN_SIMULATION_SA || userRequest == dfl::common::Options::Request::RUN_SIMULATION_NSA) {
+    params.setSimulationKind(dfl::inputs::Configuration::SimulationKind::SECURITY_ANALYSIS);
 
-    if (mpiContext.isRootProc()) {
-      LOG(info, SimulationEnded, context->basename(), elapsed(timeSimuStart));
-      DYN::Trace::info(dfl::common::Log::dynaflowLauncherLogTag) << " ============================================================ " << DYN::Trace::endline;
-      LOG(info, DFLEnded, context->basename(), elapsed(timeStart));
+    if (userRequest == dfl::common::Options::Request::RUN_SIMULATION_NSA) {
+      boost::filesystem::path steadyStateNetwork{outputDir};
```

Comment:
We have a utility method to do this in Dynawo (std::string absolute(const std::string & name, const std::string & rootName = "."))